File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/invocation/entity/DubboParamValue.java
package org.javamaster.invocationlab.admin.service.invocation.entity;

import lombok.Getter;
import lombok.Setter;

import java.util.ArrayList;
import java.util.List;

/**
 * @author yudong
 */
@Getter
public class DubboParamValue implements RpcParamValue {
    private final List<String> paramTypeNames = new ArrayList<>();
    private final List<Object> paramValues = new ArrayList<>();
    @Setter
    private String registry;
    @Setter
    private String dubboUrl = "dubbo://ip";
    @Setter
    private boolean useDubbo = false;

    public void addParamTypeName(String typeName) {
        paramTypeNames.add(typeName);
    }

    public void addParamValue(Object paramValue) {
        paramValues.add(paramValue);
    }

}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/invocation/entity/DubboParamValue.kt
package org.javamaster.invocationlab.admin.service.invocation.entity

/**
 * @author yudong
 */

class DubboParamValue : RpcParamValue {
    var paramTypeNames: MutableList<String> = mutableListOf()
    var paramValues: MutableList<Any> = mutableListOf()
    var registry: String? = null
    var dubboUrl = "dubbo://ip"
    var isUseDubbo = false

    fun addParamTypeName(typeName: String) {
        paramTypeNames.add(typeName)
    }

    fun addParamValue(paramValue: Any) {
        paramValues.add(paramValue)
    }
}
